homeassistant/components/proximity/coordinator.py

```diff
@@ -74,15 +74,14 @@ def __init__(
             "dir_of_travel": DEFAULT_DIR_OF_TRAVEL,
             "nearest": DEFAULT_NEAREST,
         }
-
         self.state_change_data: StateChangedData | None = None
 
     async def async_check_proximity_state_change(
         self, entity: str, old_state: State | None, new_state: State | None
     ) -> None:
         """Fetch and process state change event."""
         if new_state is None:
```

Comment:
this check is "old code", further I'm not sure in which circumstances `new_state` could be `None`, therefore I've no clue how to write a test for it

Reply:
i get it ... `new_state` is `None` in case of an entity is removed